test/spec/runners/operations.ex

```diff
@@ -0,0 +1,221 @@
+
+      "withdrawals" ->
+        assert_process_withdrawal(pre, operation, post)
+
+      "bls_to_execution_change" ->
+        assert_process_bls_to_execution_change(pre, operation, post)
+
+      "deposit_receipt" ->
+        assert_process_deposit_receipt(pre, operation, post)
+
+      handler ->
+        raise "Unknown case: #{handler}"
+    end
+  end
+
+  def assert_process_attestation(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_attester_slashing(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_block_header(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_deposit(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_proposer_slashing(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_voluntary_exit(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_sync_aggregate(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_execution_payload(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_withdrawal(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_bls_to_execution_change(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  def assert_process_deposit_receipt(_pre, _operation, _post, _data \\ "none") do
+    # TODO
+  end
+
+  @doc """
+  Prepares the data for the tests: build_path() |> decompress() |> deserialize()
+  """
+  def prepare_test(case_dir, handler, name) do
+    build_file_path(case_dir, name) |> decompress() |> deserialize(handler, name)
+  end
+
+  @doc """
+  Snappy decompression of the files
+  """
+  def decompress(file_path) do
+    if File.exists?(file_path) do
+      compressed = File.read!(file_path)
+      {:ok, decompressed} = :snappyer.decompress(compressed)
+      {:ok, decompressed}
+    else
+      {:ok, "no post"}
+    end
+  end
+
+  @doc """
+  Deserialization of the files
+  """
+  def deserialize({:ok, "no post"}, _, _), do: {:ok, "no post"}
+
+  def deserialize({:ok, serialized}, _handler, "pre") do
+    Ssz.from_ssz(serialized, SszTypes.BeaconState, @config)
+  end
+
+  def deserialize({:ok, serialized}, _handler, "post") do
+    Ssz.from_ssz(serialized, SszTypes.BeaconState, @config)
+  end
+
+  def deserialize({:ok, serialized}, handler, _) do
+    Ssz.from_ssz(serialized, resolve_type_from_handler(handler), @config)
+  end
+
+  def build_file_path(case_dir, name) do
+    "#{case_dir}/#{name}.ssz_snappy"
+  end
+
+  @doc """
+  Each handler has associated types, hence we resolve each type that has to be used from each handler
+  """
+  def resolve_type_from_handler(handler) do
+    case Map.get(@type_map, handler) do
+      nil -> raise "Unknown case #{handler}"
+      type -> Module.concat(SszTypes, type)
+    end
```

Comment:
[`Map.fetch!`](https://hexdocs.pm/elixir/1.12.3/Map.html#fetch!/2) can also be used for this

Reply:
Would it be better using Map.fetch! ?

Reply:
I think it'd be worse, because the message is less clear (i.e. "key ... not found in: %{...}").